src/main/java/com/cburch/logisim/fpga/gui/HdlColorRenderer.java

```diff
@@ -37,52 +40,89 @@ public HdlColorRenderer() {
   }
 
   @Override
-  public Component getTableCellRendererComponent(
-      JTable table, Object Info, boolean isSelected, boolean hasFocus, int row, int column) {
-    /* we have a difference between the first row and the rest */
-    if (row == 0) {
-      String value = (String) Info;
-      boolean passive = value.equals(NO_SUPPORT_STRING);
-      Color newColor = (passive) ? Color.red : Color.green;
-      if (value.equals(UNKNOWN_STRING)) newColor = table.getGridColor();
-      if (column == 1) setBackground(newColor);
-      setForeground(Color.black);
-      if (column == 0) setText(value);
-      else {
-        if (value.equals(NO_SUPPORT_STRING)) setText(S.get("FPGANotSupported"));
-        else if (value.equals(SUPPORT_STRING)) setText(S.get("FPGASupported"));
-        else setText(S.get("FPGAUnknown"));
-      }
+  public Component getTableCellRendererComponent(JTable table, Object infoObj, boolean isSelected,
+                                                 boolean hasFocus, int rowIdx, int columnIdx) {
+    // Some sanity checks to avoid NPE
+    final var value = (infoObj != null) ? (String) infoObj : "???";
```

Comment:
Is this something likely to occur, i.e., handling a typical runtime error or is this a precondition warranting the use of an `assert`?

Reply:
Technically speaking it should not happen, yet this is allowed value anyway, hence the fallback. 

As for `assert`, I explained why no condition warrants use of in production code [here](https://github.com/logisim-evolution/logisim-evolution/pull/1581#discussion_r1015140662).